### PR TITLE
Clear incorrectly reported errors in cms_io

### PR DESCRIPTION
**Pull request by faramir-dev**
##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/cmsapitest.c [hunk 1]
```diff
@@ -288,7 +289,38 @@ static int test_d2i_CMS_bio_NULL(void)
...
+      || !TEST_ptr(cms = d2i_CMS_bio(bio, NULL)))
+      goto end;
+
+    if (!TEST_ptr(err_bio = BIO_new(BIO_s_mem())))
+        goto end;
+    ERR_print_errors(err_bio);
+    (void)BIO_get_mem_data(err_bio, &err_buf);
+
+    if (!TEST_ptr_null(err_buf))
+        goto end;
```
Comment (t8m) [inline test/cmsapitest.c:312]: It would be much easier to just do `if (!TEST_int_eq(ERR_peek_error(), 0))` here instead of printing out the errors.

**Review by t8m — APPROVED** [review]

File: crypto/cms/cms_io.c [hunk 2]
```diff
@@ -42,7 +42,11 @@ CMS_ContentInfo *d2i_CMS_bio(BIO *bp, CMS_ContentInfo **cms)
                               ossl_cms_ctx_get0_libctx(ctx),
                               ossl_cms_ctx_get0_propq(ctx));
     if (ci != NULL)
+    {
```
Comment (slontis) [inline crypto/cms/cms_io.c:45]: Coding style for if blocks should be
````
if (...) {
.....
}

File: crypto/cms/cms_io.c [hunk 3]
```diff
@@ -105,7 +109,11 @@ CMS_ContentInfo *SMIME_read_CMS_ex(BIO *bio, int flags, BIO **bcont,
                                                ossl_cms_ctx_get0_libctx(ctx),
                                                ossl_cms_ctx_get0_propq(ctx));
     if (ci != NULL)
+    {
```
Comment (slontis) [inline crypto/cms/cms_io.c:112]: apply same change here..

**Review by slontis — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you.

**Pull request closed without merging**